pelican/utils.py

```diff
@@ -360,7 +360,7 @@ def path_to_url(path):
         return '/'.join(split_all(path))
 
 
-def truncate_html_words(s, num, end_text='...'):
+def truncate_html_words(s, num, end_text=' ...'):
```

Comment:
Shouldn't the default value of `end_text` be `self.settings['SUMMARY_END_TEXT']`? I personally don't like hardcoding the default value again if we already have a `self.settings` for it.

Reply:
I would argue that we should not bind an utility function to a specific use-case or setting and that having the hardcoded default in the utility function is fine.  I am currently working on a PR that implements this functionality, please see #2710 .